sources/web/datalab/jupyter.ts

```diff
@@ -81,67 +88,78 @@ function pipeOutput(stream: NodeJS.ReadableStream, port: number, error: boolean)
  * Starts the Jupyter server, and then creates a proxy object enabling
  * routing HTTP and WebSocket requests to Jupyter.
  */
-function createJupyterServer(userId: string): JupyterServer {
+function createJupyterServer(userId: string, resolved: (server: JupyterServer)=>void, rejected:()=>void) {
```

Comment:
Can we move the "resolved" and "rejected" logic into this function? We don't have different "resolved" or "rejected" functions anyway.